go/src/net/ConnectionPool.go

```diff
@@ -0,0 +1,235 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package nebulaNet
+
+import (
+	"container/list"
+	"errors"
+	"fmt"
+	"sync"
+
+	graph "github.com/vesoft-inc/nebula-clients/go/nebula/graph"
+	conf "github.com/vesoft-inc/nebula-clients/go/src/conf"
+	data "github.com/vesoft-inc/nebula-clients/go/src/data"
+)
+
+type RespData struct {
+	Resp *graph.ExecutionResponse
+	Err  error
+}
+
+type ConnectionPool struct {
+	idleConnectionQueue   list.List
+	activeConnectionQueue list.List
+	addresses             []data.HostAddress
+	conf                  conf.PoolConfig
+	isConnectionInUse     map[*Connection]bool
```

Comment:
you don't need isConnectionInUse, if the connection in activeConnectionQueue, it means it's used, you can make activeConnectionQueue as map